presto-main/src/main/java/com/facebook/presto/server/security/LdapFilter.java

```diff
@@ -106,153 +130,178 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
+    {
+        Optional<String> pattern = genericLdapBinder.getGroupAuthorizationSearchPattern();
+        if (!pattern.isPresent()) {
+            return;
+        }
+
         checkState(serverConfig.getUserBaseDistinguishedName() != null, "Base distinguished name (DN) for user %s is null", user);
 
-        try {
-            String searchFilter = searchFilterPattern.replaceAll("\\$\\{USER\\}", user);
-            GroupAuthorizationFilter groupFilter = new GroupAuthorizationFilter(serverConfig.getUserBaseDistinguishedName(), searchFilter);
+        String searchFilter = pattern.get().replaceAll("\\$\\{USER\\}", user);
+        String userBaseDistinguishedName = serverConfig.getUserBaseDistinguishedName();
+        String groupName = firstNonNull(serverConfig.getLdapServerType().equals(GENERIC) ? serverConfig.getGroupDistinguishedName() : "", "");
 
-            CacheLoader<GroupAuthorizationFilter, Boolean> loader = new CacheLoader<GroupAuthorizationFilter, Boolean>()
-            {
-                @Override
-                public Boolean load(@NotNull GroupAuthorizationFilter groupFilter)
-                        throws NamingException
-                {
-                    return groupFilter.authorize(context);
+        SearchControls searchControls = new SearchControls();
+        searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+        boolean authorized;
+        NamingEnumeration<SearchResult> search = null;
+        try {
+            search = context.search(userBaseDistinguishedName, searchFilter, searchControls);
+            authorized = search.hasMoreElements();
+        }
+        catch (NamingException e) {
+            throw new AuthenticationException(INTERNAL_SERVER_ERROR, "Authentication failed", e);
+        }
+        finally {
+            if (search != null) {
+                try {
+                    search.close();
+                }
+                catch (NamingException ignore) {
                 }
-            };
-            LoadingCache groupFilterCache = CacheBuilder.newBuilder().expireAfterWrite(ldapCacheTtl, MILLISECONDS).build(loader);
-
-            String groupNameIfPresent = firstNonNull(serverConfig.getLdapServerType().equals(GENERIC) ? null : serverConfig.getGroupDistinguishedName(), "");
-            if (Boolean.FALSE.equals(groupFilterCache.get(groupFilter))) {
-                String errorMessage = format("User %s not a member of the group %s", user, groupNameIfPresent);
-                log.debug("Authorization failed for user. " + errorMessage);
-                throw new RuntimeException("Unauthorized user: " + errorMessage);
             }
-            log.debug("Authorization succeeded for user %s in group %s", user, groupNameIfPresent);
         }
-        catch (ExecutionException e) {
-            throw Throwables.propagate(e);
+
+        if (!authorized) {
+            String message = format("Unauthorized user: User %s not a member of the group %s", user, groupName);
```

Comment:
A little tangential to your changes, but can the group filter ever fail if the groupName is ""? It seems weird to be doing the LDAP group search if there's no group name specified.